### PR TITLE
Refactor visualizer panel classes and event handling

### PR DESCRIPTION
**Pull request by k3vnchen**
Major credit goes to [Printrun](https://github.com/kliment/Printrun), specifically its [`panel.py`](https://github.com/kliment/Printrun/blob/master/printrun/gl/panel.py) and [`stlview.py`](https://github.com/kliment/Printrun/blob/master/printrun/stlview.py). Functionally, nothing has changed (save for the minor background color change), but this is a refactoring of how our classes and event bindings are structured. It should be a lot easier to add more features to the visualizer panel from here on out. Here's a list of things changed:
- Created `camview.py`, where `Camera` and `Camera3D` classes have been moved to (`canvas.py` and `camview.py` is based off of Printrun's `panel.py` and `stlview.py`, respectively)
- Moved all event bindings except `EVT_ERASE_BACKGROUND`, `EVT_SIZE`, and `EVT_PAINT` from `CanvasBase` to `Canvas`
  - Removed `EVT_LEFT_DOWN`, `EVT_LEFT_UP`, and `EVT_MOTION` event bindings, instead favoring a general `EVT_MOUSE_EVENTS` with further left/right click and drag processing within `move()`
  - Mouse panning (for rotating), as a result, no longer uses `OnMouseDown()`, `OnMouseUp()`, or `OnMouseMotion()`, but only `move()` and `handle_rotation()`
- Panel resize handling and zoom perspective updating is now handled in `OnReshape()` rather than `OnDraw()`; as a result, `OnDraw()` and `draw_objects()` is massively simplified
- Added some `__init__` variables, unused for now but will be for trackball rotation (next steps)
  - `self.width` and `self.height` see a resurrection! Life is just easier with them
- Copied over Printrun `panel.py`'s `mouse_to_3d`, `mouse_to_ray`, and `mouse_to_plane`, not in use currently, but will be used for future navigation improvements
- Changed window background color and grid color to be more pleasing to the eye
- All `__init__`s for wx windows have `*args` and `**kwargs` arguments added, as per the [wxPython style guide](https://wiki.wxpython.org/wxPython%20Style%20Guide)

### CODE REVIEW

**Comment by k3vnchen** [comment]
> Please add comments in camview.py and canvas.py.

Done! Let me know if more needs to be done.

File: components/camview.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        if event.Dragging() and event.LeftIsDown():
+            self.handle_rotation(event)
+        # elif event.Dragging() and event.RightIsDown():
+            # self.handle_translation(event)
+        elif event.ButtonUp(wx.MOUSE_BTN_LEFT):
+            if self.initpos is not None:
+                self.initpos = None
+        elif event.ButtonUp(wx.MOUSE_BTN_RIGHT):
+            if self.initpos is not None:
+                self.initpos = None
```
[the remark this answers is not in the document]
Comment (k3vnchen) [reply components/camview.py:67]: `move()` was taken from Printrun's `stlview.py` - I tested it, and there doesn't seem to be any functional difference between having the two separate conditionals doing the same thing versus having a general `event.ButtonUp()` (which captures all button up events). Honestly not sure why it's set this way, but I guess it doesn't hurt?

[the remark this answers is not in the document]
Comment (k3vnchen) [reply components/camview.py:67]: Or would you rather we combine it?

[the remark this answers is not in the document]
Comment (parksong22) [reply components/camview.py:67]: Well, if there is no difference, I would rather removed the duplication of code. Can you please test with general `event.ButtonUp()` and see if that makes any difference? If not, I prefer to have just one ButtonUp event.

[the remark this answers is not in the document]
Comment (k3vnchen) [reply components/camview.py:64]: Just tested - no difference. Merged them and pushed changes

**Pull request merged**